packages/react-composites/src/composites/CallComposite/CallComposite.tsx

```diff
@@ -97,13 +98,25 @@ type MainScreenProps = {
 
 const MainScreen = (props: MainScreenProps): JSX.Element => {
   const { callInvitationUrl, onRenderAvatar, onFetchAvatarPersonaData, onFetchParticipantMenuItems } = props;
-  const page = useSelector(getPage);
+  let page = useSelector(getPage);
 
   const adapter = useAdapter();
   const locale = useLocale();
 
   let pageElement: JSX.Element;
 
+  if (props.options?.endCallScreen === false) {
+    // keep the user on the call screen knowing that the developer will change up the UI at their earliest convenience
+    if (
+      page === 'leftCall' ||
+      page === 'joinCallFailedDueToNoNetwork' ||
+      page === 'accessDeniedTeamsMeeting' ||
+      page === 'removedFromCall'
+    ) {
+      page = 'call';
```

Comment:
quick note: if going down this route we should use page's last render page. e.g. if I denied access via teams lobby it shouldn't try and render a call page
```
const page = useRef<Page>();
let previousPage = page.current;
page.current = useSelector(getPage);
```

Reply:
I worry about this route as the props and selectors may rely on the call state being correctly in a call - we may get UI changing on the call page for split second when props are changed (e.g. I presume we don't have any remote participants any more when we render the call page after onCallEnded occurs)

Reply:
Alternative way to handle emitting callEnded when the state is transitioning: #2201

Reply:
I admit I primarily tested out this feature with leftCall which worked out well. I think we should definitely validate some of these other scenarios.

I like this last page concept.